sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -175,10 +175,22 @@ class AstBuilder extends DataTypeAstBuilder with SQLConfHelper with Logging {
         .map { s =>
           SingleStatement(parsedPlan = visit(s).asInstanceOf[LogicalPlan])
         }.getOrElse {
-          visit(ctx.beginEndCompoundBlock()).asInstanceOf[CompoundPlanStatement]
+          visitChildren(ctx).asInstanceOf[CompoundPlanStatement]
         }
     }
 
+  override def visitIfElseStatement(ctx: IfElseStatementContext): IfElseStatement = {
+    IfElseStatement(
+      conditions = ctx.booleanExpression().asScala.toList.map(boolExpr => withOrigin(boolExpr) {
```

Comment:
can `conditions` be `Seq[Expression]`?

Reply:
is there an easy way to evaluate boolean expression (except what we did here - wrapping inside of a `Project`)?

the reason why we did it like this, as I explained in the other comment, is because we want to be able to iterate through the condition as well, and this approach is complicated a bit if the condition is not an instance of `SingleStatement`.

if we need to wrap expression into something similar to what we did here to evaluate it properly, I don't see a reason not to do it immediately and then use the logical plan to construct the `SingleStatement`, as we did here.

but, as I said, depends a bit on whether there is a better way to evaluate boolean expressions?